File: Client_Simple/Sources/Couleur.cpp
#include "../Headers/Couleur.h"

const QString Couleur::strCouleur[] = { "BLACK", "BLUE", "RED", "GREEN", "YELLOW", "CYAN" };

QString Couleur::getCouleur(const int i)
{
    if (i < 0 || i > NB__)
    {
		abort();
	}
	return strCouleur[i];
}

Couleur::Couleurs Couleur::getNameCouleur(const QString s)
{
    if(s == "BLACK")
        return Couleur::BLACK;
    else if(s == "BLUE")
        return Couleur::BLUE;
    else if(s == "RED")
        return Couleur::RED;
    else if(s == "GREEN")
        return Couleur::GREEN;
    else if(s == "BLUE")
        return Couleur::BLUE;
    else if(s == "YELLOW")
        return Couleur::YELLOW;
    else if(s == "CYAN")
        return Couleur::CYAN;
    else
        return Couleur::BLACK;
}

File: ServeurGeo2D/src/Construction/Couleur.java
package Construction;

public class Couleur {
	private int r;
	private int g;
	private int b;
	
	Couleur(String s)
	{
		if(s.equalsIgnoreCase("BLUE"))
		{
			this.setR(0);
			this.setG(0);
			this.setB(255);
		}
		else if(s.equalsIgnoreCase("RED"))
		{
			this.setR(255);
			this.setG(0);
			this.setB(0);
		}
		else if(s.equalsIgnoreCase("GREEN"))
		{
			this.setR(0);
			this.setG(255);
			this.setB(0);
		}
		else if(s.equalsIgnoreCase("YELLOW"))
		{
			this.setR(255);
			this.setG(255);
			this.setB(0);
		}
		else if(s.equalsIgnoreCase("CYAN"))
		{
			this.setR(0);
			this.setG(255);
			this.setB(255);
		}
		else
		{
			this.setR(0);
			this.setG(0);
			this.setB(0);
		}
	}
	
	public int getR()
	{
		return r;
	}
	
	public int getG()
	{
		return g;
	}
	
	public int getB()
	{
		return b;
	}
	
	public void setR(int red)
	{
		if(red >= 0 && red <= 255)
			r = red;
	}
	
	public void setG(int green)
	{
		if(green >= 0 && green <= 255)
			g = green;
	}
	
	public void setB(int blue)
	{
		if(blue >= 0 && blue <= 255)
			b = blue;
	}
	
}
